scripts/tests/dfx_tests.sh

```diff
@@ -0,0 +1,74 @@
+#!/bin/bash
+
+export ORD_BITCOIN_RPC_USERNAME=ic-btc-integration
+export ORD_BITCOIN_RPC_PASSWORD="QPQiNaph19FqUsCrBRN0FII7lyM26B51fAMeBQzCb-E="
+LOGFILE=./target/dfx_tests.log
+
+setup_docker() {
+    PREV_PATH=$(pwd)
+    cd btc-deploy/
+    docker compose up -d --build
+    cd $PREV_PATH
+}
+
+stop_docker() {
+    PREV_PATH=$(pwd)
+    cd btc-deploy/
+    docker compose down
+    cd $PREV_PATH
+}
+
+WITH_DOCKER="0"
+if [ "$1" == "--docker" ]; then
+    WITH_DOCKER="1"
+    shift
+fi
+
+# set dfxvm to use the correct version and use docker
+if [ "$1" == "--github-ci" ]; then
+    shift
+    WITH_DOCKER="1"
+    dfxvm default 0.16.1
```

Comment:
1. why do we need a fixed version of dfx? 
2. use `--docker` to set the WITH_DOCKER var